src/js/_enqueues/admin/user-profile.js

```diff
@@ -332,6 +340,75 @@
 		}
 	}
 
+	/**
+	 * Binds Caps Lock detection to a given password input field.
+	 *
+	 * @param {jQuery} $input The password input field.
+	 */
+	function bindCapsLockWarning( $input ) {
+		var $capsWarning = $( '#caps-warning' ),
+			capsLockOn   = false;
+
+		$input.on( 'keydown', function( e ) {
+			var ev = e.originalEvent || e;
+
+			// Skip CapsLock key itself.
+			if ( ev.key === 'CapsLock' ) {
+				if (capsLockOn) {
+					capsLockOn = false;
+					$capsWarning.hide();
+				}
+				return;
+			}
+
+			// Skip if key is not a printable character.
+			// Key length > 1 usually means non-printable (e.g., "Enter", "Tab").
+			if ( ev.ctrlKey || ev.metaKey || ev.altKey || ! ev.key || ev.key.length !== 1 ) {
+				return;
+			}
+
+			var state = isCapsLockOn( ev );
+
+			// Only react when the state changes.
+			if ( state !== capsLockOn ) {
+				capsLockOn = state;
+
+				if ( capsLockOn ) {
+					$capsWarning.show();
+					wp.a11y.speak( __( 'Caps lock is on.' ) );
+				} else {
+					$capsWarning.hide();
+				}
+			}
+		} );
+
+		$input.on( 'blur', function() {
+			capsLockOn = false;
+			$capsWarning.hide();
+		} );
+	}
+
+	/**
+	 * Determines if Caps Lock is currently enabled.
+	 *
+	 * On macOS Safari and Firefox, the native warning is preferred,
+	 * so this function returns false to suppress custom warnings.
+	 *
+	 * @param {KeyboardEvent} e The keydown event object.
+	 *
+	 * @return {boolean} True if Caps Lock is on, false otherwise. 
+	 */
+	function isCapsLockOn( e ) {
+		// Skip warning on macOS Safari + Firefox (they show native indicators).
+		if ( isMac && ( isSafari || isFirefox ) ) {
+			return false;
+		}
+
+		if ( typeof e.getModifierState === 'function' ) {
+			return e.getModifierState( 'CapsLock' );
+		}
```

Comment:
On second thought, there's no need for the function exists check. This method is available in all supported browsers: https://caniuse.com/mdn-api_mouseevent_getmodifierstate

```suggestion
		return e.getModifierState( 'CapsLock' );
```